### PR TITLE
Stepper component

### PR DESCRIPTION
**Pull request by edoardocavazza**
**⚠️ THIS ELEMENT IS A SNOWFLAKE, DO NOT MERGE THE PR ⚠️**

This PR introduces the new Stepper component. At the moment, the abstract link is missing because this element is still a proposal.

### CODE REVIEW

**Comment by wanbinkimoon** [comment]
@edoardocavazza why a PR instead of a separated branch such as `snowflake/stepper`? 
Do you think this component will be merged in the DS?

**Comment by edoardocavazza** [comment]
As far as I know, there is draft of proposal for this component that is going to be submitted soon.

**Comment by ufollettu** [comment]
I report unexpected behavior: if I pass a `z-stepper-item` into the `z-stepper` slot, the display is correct, but if I pass, for example, a link tag that wraps the `z-stepper-item`, the separators are no longer visible (see attached image).
image

**Comment by edoardocavazza** [comment]
Hi @ufollettu, can you give a try to the new version with `href` attribute support?

File: src/components/indicators/z-stepper-item/styles.css [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+}
+
+:host([disabled]) .indicator {
+  color: var(--text-grey-800);
+  background: var(--bg-grey-200);
+  border-color: var(--text-grey-800);
+}
+
+/* Mobile breakpoint */
+@media only screen and (max-width: 768px) {
```
Comment (vpasquino) [inline src/components/indicators/z-stepper-item/styles.css:93]: CSS should be mobile first (if you create components with `yarn generate` the file is created with breakpoints for instance `@media only screen and (min-width: 768px) {}`). 
Also, 768px should still be tablet.

File: src/components/indicators/z-stepper-item/styles.css [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+}
+
+:host([pressed]:not([disabled])) .indicator {
+  color: var(--text-white);
+  background: var(--myz-blue);
+  border-color: var(--myz-blue);
+  box-shadow: none;
+}
+
+:host([disabled]) .indicator {
```
Comment (vpasquino) [inline src/components/indicators/z-stepper-item/styles.css:86]: From abstract proposal: `--bg-grey-700`

File: src/components/indicators/z-stepper-item/styles.css [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+  border-color: var(--myz-blue-dark);
+}
+
+:host(:not([disabled]):not([pressed]):focus-within) .indicator {
+  color: var(--myz-blue);
+  border-color: var(--myz-blue);
+  box-shadow: 0px 0px 2px 2px var(--accent-dark);
+}
+
+:host([pressed]:not([disabled])) .indicator {
```
Comment (vpasquino) [inline src/components/indicators/z-stepper-item/styles.css:85]: From abstract proposal: `--myz-blue-dark`

File: src/components/indicators/z-stepper-item/styles.css (hunk 1 above)
Comment (edoardocavazza) [reply src/components/indicators/z-stepper-item/styles.css]: The comment was wrong but it seems this is an old version of the media query 🤔

Comment (vpasquino) [reply src/components/indicators/z-stepper-item/styles.css]: I'm not sure :see_no_evil:

**Comment by vpasquino** [comment]
Looks perfect! :)

Tests are broken though.

File: src/components/indicators/z-stepper-item/styles.css (hunk 1 above)
Comment (edoardocavazza) [reply src/components/indicators/z-stepper-item/styles.css]: The media query should had changed in this commit: https://github.com/ZanichelliEditore/design-system/commit/1c89f7cdbee714ea1fb72ba63c685671296a538e#diff-3922f519e39d8c8f05e9d120afa4caf0R99

Anyway, I think it's ok now :)

Comment (vpasquino) [reply src/components/indicators/z-stepper-item/styles.css]: Haha the commit is old but I think you pushed it this morning. Great! :smile:

**Pull request merged**